sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -85,9 +85,11 @@ private[sql] object DataSourceStrategy extends Strategy with Logging {
         s"Selected $selected partitions out of $total, pruned $percentPruned% partitions."
       }
 
+      // need to add projections from combineFilters in
+      val combineProjections = projects.toSet.union(combineFilters.flatMap(_.references).toSet).toSeq
```

Comment:
We use use an `AttributeSet` anytime we are dealing with `AttributeReference` objects.